### DISCUSSION TITLE
Running the integrated GraphGym

### DISCUSSION THREAD

**Discussion by nilskre** [pyg-team/pytorch_geometric#7248] [Q&A] [answered]
Two questions regarding running the integrated GraphGym:
1. Is there any documentation, on how the virtual environment for GraphGym needs to be set up / which dependencies are necessary?
2. The docs state, that running on the CPU is possible by adding ```device: cpu to the *.yaml file```. Is [this file](https://github.com/pyg-team/pytorch_geometric/blob/master/graphgym/configs/example.yaml) meant? After adding it there it seems to work (Log: ```GPU available: True (cuda), used: False```). But running a single experiment still fails with ```RuntimeError: CUDA unknown error - this may be due to an incorrectly set up environment, e.g. changing env variable CUDA_VISIBLE_DEVICES after program start. Setting the available devices to be zero.```

Thanks in advance!

**Comment by rusty1s** [comment]
1. Necessary dependencies can be installed via `pip install torch-geometric[graphgym]`.
2. Let me take a look.

**Comment by rusty1s** [comment, marked as answer]
I have set `accelerator: cpu` in the `yml` file, and it runs fine for me. I can confirm that the data is still on the CPU. Can you share some more information where it crashes for you exactly?

**Comment by nilskre** [comment]
Thanks a lot for this fast and helpful response!
I can now confirm, that with ```accelerator: cpu``` in the yml file everything works as expected. Thanks! 
Could it be the case, that the [docs](https://pytorch-geometric.readthedocs.io/en/latest/advanced/graphgym.html#basic-usage) about this are outdated? They state ```Run GraphGym with CPU backend: GraphGym supports CPU backend as well – you only need to add the line device: cpu to the *.yaml file.``` (this is not working).

**Reply by rusty1s** [reply]
Yes, indeed. Will fix.

**Reply by rusty1s** [reply]
Fixed in https://github.com/pyg-team/pytorch_geometric/pull/7274.
